### PR TITLE
[254] Add delete from model tool on containment edges

### PR DESCRIPTION
**Pull request by gdaniel**
Bug: https://github.com/eclipse-syson/syson/issues/254

### CODE REVIEW

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractDefinitionOwnedUsageEdgeDescriptionProvider.java [hunk 1]
```diff
@@ -127,7 +128,18 @@ private EdgeStyle createEdgeStyle() {
     @Override
     protected boolean isDeletable() {
         // composition edges are not deletable
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractDefinitionOwnedUsageEdgeDescriptionProvider.java:130]: Please remove this comment as it is not true anymore

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java [hunk 2]
```diff
@@ -109,8 +110,18 @@ private EdgeStyle createEdgeStyle() {
 
     @Override
     protected boolean isDeletable() {
-        // composition edges are not deletable
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java:112]: same

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java [hunk 3]
```diff
@@ -496,7 +496,10 @@ public Element reconnnectTargetCompositionEdge(Element self, Element oldTarget,
         } else {
             var oldContainer = oldTarget.eContainer();
             if (newTarget instanceof Usage && oldContainer instanceof FeatureMembership featureMembership) {
-                // move the old target to the innermost package
+                // Move the old target to the innermost package
+                // We don't use moveToClosestContainingPackage here because we are moving oldTarget in the closest
+                // containing package
+                // and updating newTarget at the same time.
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java:502]: the third line should probably be append on the second line

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java [hunk 4]
```diff
@@ -514,6 +517,27 @@ public Element reconnnectTargetCompositionEdge(Element self, Element oldTarget,
         return self;
     }
 
+    /**
+     * Moves the provided {@code element} in the closest containing package.
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java:521]: element -> usage

Comment (gdaniel) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java]: done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java (hunk 3 above)
Comment (gdaniel) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java]: done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java (hunk 2 above)
Comment (gdaniel) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractUsageNestedUsageEdgeDescriptionProvider.java:112]: done

File: backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractDefinitionOwnedUsageEdgeDescriptionProvider.java (hunk 1 above)
Comment (gdaniel) [reply backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/edges/AbstractDefinitionOwnedUsageEdgeDescriptionProvider.java]: done

**Pull request merged**